jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -139,6 +140,31 @@ def _parse_image(app, file_obj, data_label, show_in_viewer, ext=None):
         if show_in_viewer:
             app.add_data_to_viewer("viewer-1", data_label)
 
+    if len(app.data_collection) <= 1:  # No need to link, we are done.
+        return
+
+    # Auto-link data by pixels
+
+    links_list = []
+    refdata = app.data_collection[0]  # Link with first one
+    ids0 = refdata.pixel_component_ids
+    ndim_range = range(refdata.ndim)
+
+    for data in app.data_collection[1:]:
+        ids1 = data.pixel_component_ids
+        try:
+            new_links = [LinkSame(ids0[i], ids1[i]) for i in ndim_range]
+        except Exception as e:
+            # TODO: Is it better to just throw exception and crash?
+            app.hub.broadcast(SnackbarMessage(
+                f"Error linking '{data.label}' to '{refdata.label}': "
+                f"{repr(e)}", color="warning", timeout=8000, sender=app))
+            continue
+        links_list += new_links
+
+    with app.data_collection.delay_link_manager_update():
+        app.data_collection.set_links(links_list)
```

Comment:
Why does the block above Not use the new link_data function you have written?

Reply:
Only because I don't know how to access the `Imviz` class object from the parser, which only takes `app`. If you know how, please let me know.

Reply:
What if you wrote add_links as a function not a method which could then be called from both the imviz class or the parser?

Reply:
Maybe, but we're at crunch time, so if this is good enough, I'd rather make it cleaner as "future work". 😬 I really want to get this merged so we can go back to the pan-zoom lag problem.

Reply:
Your suggestion here is now part of #909.